src/stream/slpFolderStream.ts

```diff
@@ -86,9 +88,15 @@ export class SlpFolderStream extends RxSlpStream {
       this.readStream.unpipe(this);
       this.readStream.done();
       this.readStream = null;
+      this.watcher?.close();
+      this.watcher = null;
```

Comment:
```suggestion
     if (this.watcher) {
       this.watcher.close();
       this.watcher = null;
     }
```
I would prefer an explicit check here.